### PR TITLE
docs: fix event points descriptions

### LINKED ISSUE #8049: Event Point `pre_system` and `post_system` are different from the docs
Are these bugs or not?

https://codeigniter4.github.io/CodeIgniter4/extending/events.html#event-points
- `pre_system` Called **very early** during system execution. **Only the benchmark and events class have been loaded** at this point. No routing or other processes have happened.
- `post_system` Called **after the final rendered page is sent to the browser**, at the end of system execution after the finalized data is sent to the browser.

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes #8049

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

**Review by MGatner — COMMENTED** [review]
A couple comments

File: user_guide_src/source/installation/upgrade_4xx.rst [hunk 1]
```diff
@@ -153,6 +153,12 @@ Hooks
 - Instead of CI3's ``$hook['post_controller_constructor']`` you now use
   ``Events::on('post_controller_constructor', ['MyClass', 'MyFunction']);``, with the namespace ``CodeIgniter\Events\Events;``.
 - Events are always enabled, and are available globally.
+- The hook point ``pre_controller`` and ``post_controller`` have been removed.
```
Comment (MGatner) [inline user_guide_src/source/installation/upgrade_4xx.rst:156]: Where did this come from?

File: user_guide_src/source/installation/upgrade_4xx.rst [hunk 2]
```diff
@@ -153,6 +153,12 @@ Hooks
...
+  Use :doc:`../incoming/filters` instead.
+- The hook point ``display_override`` and ``cache_override`` have been removed.
+  Because the base methods have been removed.
+- The hook point ``post_system`` has moved just before sending the final rendered
```
Comment (MGatner) [inline user_guide_src/source/installation/upgrade_4xx.rst:160]: Did this actually move?

File: user_guide_src/source/installation/upgrade_4xx.rst (hunk 1 above)
Comment (kenjis) [reply user_guide_src/source/installation/upgrade_4xx.rst:156]: The source code, and the current docs.
```console
$ grep -R Events::trigger system/
system//Database/BaseConnection.php:                Events::trigger('DBQuery', $query);
system//Database/BaseConnection.php:            Events::trigger('DBQuery', $query);
system//Database/BaseConnection.php:        Events::trigger('DBQuery', $query);
system//Database/MigrationRunner.php:        Events::trigger('migrate', $data);
system//Database/MigrationRunner.php:        Events::trigger('migrate', $data);
system//Database/BasePreparedQuery.php:                Events::trigger('DBQuery', $query);
system//Database/BasePreparedQuery.php:            Events::trigger('DBQuery', $query);
system//Database/BasePreparedQuery.php:        Events::trigger('DBQuery', $query);
system//Test/Mock/MockEmail.php:            Events::trigger('email', $this->archive);
system//CodeIgniter.php:        Events::trigger('pre_system');
system//CodeIgniter.php:            Events::trigger('post_controller_constructor');
system//CodeIgniter.php:        Events::trigger('post_system');
system//Email/Email.php:            Events::trigger('email', $this->archive);
system//Email/Email.php:        Events::trigger('email', $this->archive);
```

File: user_guide_src/source/installation/upgrade_4xx.rst (hunk 2 above)
Comment (kenjis) [reply user_guide_src/source/installation/upgrade_4xx.rst:160]: What do you mean? In CI4, post_system is triggered before sending the response.
https://github.com/codeigniter4/CodeIgniter4/blob/cf43310eca93dc8edd7be3ffd86584971890628e/system/CodeIgniter.php#L554-L557

https://github.com/bcit-ci/CodeIgniter/blob/63d037565dd782795021dcbd3b1ca6f8c8a509e4/system/core/CodeIgniter.php#L494-L509

Comment (MGatner) [reply user_guide_src/source/installation/upgrade_4xx.rst:160]: Ohhhh I missed that this was in the CI3 transition docs! 🤦‍♂️ Sorry

**Review by MGatner — APPROVED** [review]

**Comment by kenjis** [comment]
Oh, v4.4.2 was already released!

**Comment by kenjis** [comment]
I created a new PR #8076

**Comment by MGatner** [comment]
Sorry I missed that. New PR looks good.

**Pull request merged**